misc/ballerina-bindgen/src/main/java/org/ballerinalang/bindgen/utils/BindgenFileGenerator.java

```diff
@@ -53,23 +53,19 @@ public BindgenFileGenerator(BindgenEnv env) {
 
     public SyntaxTree generate(JError jError) throws BindgenException {
         this.currentClass = jError.getCurrentClass();
-        setClassNameAlias();
-        return generateFromTemplate(jErrorTemplatePath);
+        return generateFromTemplate(jErrorTemplatePath, jError.getShortExceptionName());
     }
 
     public SyntaxTree generate(JClass jClass) throws BindgenException {
         this.jClass = jClass;
         this.currentClass = jClass.getCurrentClass();
         setClassNameAlias();
-        if (Throwable.class.isAssignableFrom(jClass.getCurrentClass())) {
-            // Generate Ballerina error bindings for Java throwable classes.
-            return generateFromTemplate(jErrorTemplatePath);
-        } else if (!env.isDirectJavaClass()) {
+        if (!env.isDirectJavaClass()) {
             // Generate Ballerina empty class bindings for dependent Java classes.
-            return generateFromTemplate(jEmptyClassTemplatePath);
+            return generateFromTemplate(jEmptyClassTemplatePath, null);
         } else {
             // Generate Ballerina class bindings for Java classes.
-            syntaxTree = generateFromTemplate(jClassTemplatePath);
+            syntaxTree = generateFromTemplate(jClassTemplatePath, null);
```

Comment:
@IrushiL Instead of passing `null` and check it internally, can't we overload a `generateFromTemplate` method to get alias from `env.getAlias`

Reply:
Done